File: android/src/main/java/com/getcapacitor/community/classes/events/RSSIUpdatedEvent.java
package com.getcapacitor.community.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;

public class RSSIUpdatedEvent extends PeerIDEvent {

    int rssi;

    public RSSIUpdatedEvent(@NonNull String peerID, int rssi) {
        super(peerID);
        this.rssi = rssi;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = super.toJSObject();

        result.put("rssi", rssi);

        return result;
    }
}

File: ios/Sources/BitchatPlugin/Classes/Events/RSSIUpdatedEvent.swift
import Foundation
import Capacitor

@objc public class RSSIUpdatedEvent: PeerIDEvent {
    let rssi: Int

    init(_ peerID: String, _ rssi: Int) {
        self.rssi = rssi

        super.init(peerID)
    }

    override public func toJSObject() -> JSObject {
        var result = super.toJSObject()

        result["rssi"] = rssi

        return result
    }
}
